src/lock_free/mwcas_descriptor.cpp

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2024 Database Group, Nagoya University
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// the corresponding header
+#include "dbgroup/atomic/mwcas/lock_free/mwcas_descriptor.hpp"
+
+// C++ standard libraries
+#include <atomic>
+#include <cstddef>
+
+// external libraries
+#include "dbgroup/lock/common.hpp"
+
+// local sources
+#include "dbgroup/atomic/mwcas/utility.hpp"
+
+namespace dbgroup::atomic::mwcas::lock_free
+{
+
+auto
+MwCASDescriptor::MwCAS()  //
+    -> bool
+{
+  const auto desc_addr = std::bit_cast<uint64_t>(this) | kMwCASFlag;
+  if (stat_.load(std::memory_order_seq_cst) == kUndecided) {
+    auto status = kSucceeded;
+    auto undecided = kUndecided;  // あまりよくない方法かも
```

Comment:
個別コメントのとおり．

```suggestion
  auto cur_stat = stat_.load(kSeqCst);
  if (cur_stat == kUndecided) {
    auto stat = kSucceeded;
```

なお， `kSeqCst` は一旦定数として定義しておいてください．最終的にフェンスを緩めるので後々消しますが，記述を短くしておくために．